widgetschema.json

```diff
@@ -0,0 +1,244 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "WidgetProperties",
+    "type": "object",
+    "required": ["id", "name", "description", "version", "properties"],
+    "additionalProperties": false,
+    "properties": {
+        "id": {
+            "type": "string",
+            "description": "Unique identifier for the widget"
+        },
+        "name": {
+            "type": "string",
+            "description": "Name of the widget"
+        },
+        "description": {
+            "type": "string",
+            "description": "Description of the widget"
+        },
+        "version": {
+            "type": "string",
+            "description": "Version of the widget"
+        },
+        "src": {
+            "type": "string",
+            "description": "URL of the widget source code"
+        },
+        "needsDataSource": {
+            "type": "boolean",
+            "description": "Indicates if the widget needs a data source"
+        },
+        "needsInstance": {
+            "oneOf": [{ "type": "boolean" }, { "$ref": "#/definitions/dataType" }]
+        },
+        "typesSupported": {
+            "type": "array",
+            "items": { "$ref": "#/definitions/dataType" }
+        },
+        "needsDocument": {
+            "type": "string"
+        },
+        "icon": {
+            "type": "object",
+            "required": ["src"],
+            "additionalProperties": false,
+            "properties": {
+                "src": {
+                    "type": "string"
+                }
+            }
+        },
+        "colspan": {
```

Comment:
Correspondences, Data Details, Page Header, Form, Payments, and View Details have it set to 12

Reply:
@bradwray Can you clarify the use of this?  Why would any widget set a `colspan`?  Seems like this is set in page configuration, not a widget property.

Reply:
@actong That is where we put colSpan to start (as a sibling of properties) and it has remained. I believe that the thinking is that it would give the widget developer the ability to set the default colSpan width for a particular widget. It has been incorporated into CDR-1128 because it was asked that any item on a page (whether a container or widget) have a colspan, but these could be set in config like you said. Also, it always appears to have the fallback value of 12 anyway. 

I dunno. What do you think? Don't enforce this one?

Reply:
Yeah, I would remove it.  I can't imagine a reason why a widget developer would set a colspan.  That's something you set when configuring the page.

Reply:
Removing it will create warnings where it is still being used